engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -102,27 +107,28 @@ object aggregates {
     override def computeOutputType(input: TypingResult): Validated[String, TypingResult]
       = Valid(Typed.genericTypeClass[java.util.Set[_]](List(input)))
 
+    override def computeStoredType(input: TypingResult): Validated[String, TypingResult]
+      = Valid(Typed.genericTypeClass[Set[_]](List(input)))
+
   }
 
   object FirstAggregator extends Aggregator {
 
-    private object Marker extends Serializable
-
-    override type Aggregate = AnyRef
+    override type Aggregate = Option[AnyRef]
 
     override type Element = AnyRef
 
-    override def zero: Aggregate = Marker
+    override def zero: Aggregate = None
 
-    override def addElement(el: Element, agg: Aggregate): Aggregate = if (agg == zero) el else agg
+    override def addElement(el: Element, agg: Aggregate): Aggregate = if (agg.isEmpty) Some(el) else agg
```

Comment:
It will break state for this aggregator, can you add some notice to migrationguide?

Reply:
:+1: